packages/notifier/src/notifier.js

```diff
@@ -152,7 +153,106 @@ export const makeNotifierKit = (...initialStateArr) => {
 };
 
 /**
- * Adaptor from async iterable to notifier.
+ * @template T
+ * @param {ERef<Subscriber<T>>} subscriberP
+ * @returns {Promise<Notifier<T>>}
+ */
+export const makeNotifierFromSubscriber = async subscriberP => {
+  const iterable = subscribeLatest(subscriberP);
+  const iterator = iterable[Symbol.asyncIterator]();
+
+  /** @type {undefined | Promise<{value, updateCount}>} */
+  let latestResultOutP;
+  /** @type {undefined | Promise<{value, updateCount}>} */
+  let nextResultOutP;
+  /** @type {UpdateCount & bigint} */
+  let latestUpdateCount = 0n;
+  let finished = false;
+
+  /**
+   * @param {Promise<IteratorResult<any, any>>} resultInP
+   * @returns {Promise<{value, updateCount}>}
+   */
+  function translateInboundResult(resultInP) {
+    return resultInP.then(
+      ({ value, done }) => {
+        // Shift the result-out promises.
+        latestResultOutP = nextResultOutP;
+        nextResultOutP = undefined;
+
+        if (done) {
+          finished = true;
+
+          // Final results have undefined updateCount.
+          return harden({ value, updateCount: undefined });
+        }
+
+        latestUpdateCount += 1n;
+        return harden({ value, updateCount: latestUpdateCount });
+      },
+      rejection => {
+        // Shift the result-out promises.
+        latestResultOutP = nextResultOutP;
+        nextResultOutP = undefined;
+
+        finished = true;
+        throw rejection;
+      },
+    );
+  }
+
+  /**
+   * @template T
+   * @type {BaseNotifier<T>}
+   */
+  const baseNotifier = Far('baseNotifier', {
+    getUpdateSince(updateCount = -1n) {
+      assert(
+        updateCount <= latestUpdateCount,
+        'argument must be a previously-issued updateCount.',
+      );
+
+      // If we don't yet have a promise for the latest outbound result, create it.
+      if (!latestResultOutP) {
+        nextResultOutP = translateInboundResult(iterator.next());
+        latestResultOutP = nextResultOutP;
+      }
+
+      let resultP =
+        updateCount < latestUpdateCount || finished
+          ? latestResultOutP
+          : nextResultOutP;
+
+      if (!resultP) {
+        nextResultOutP = translateInboundResult(iterator.next());
+        resultP = nextResultOutP;
+      }
+
+      // Each returned promise is unique.
+      return resultP.then();
+    },
+  });
+
+  /** @type {Notifier<T>} */
+  const notifier = Far('notifier', {
+    ...makeAsyncIterableFromNotifier(baseNotifier),
+    ...baseNotifier,
+
+    /**
+     * Use this to distribute a Notifier efficiently over the network,
+     * by obtaining this from the Notifier to be replicated, and applying
+     * `makeNotifier` to it at the new site to get an equivalent local
+     * Notifier at that site.
+     */
+    getSharableNotifierInternals: () => baseNotifier,
+    getStoreKey: () => harden({ notifier }),
+  });
+  return notifier;
+};
+harden(makeNotifierFromSubscriber);
+
+/**
+ * Deprecated adaptor from async iterable to notifier.
```

Comment:
https://www.typescriptlang.org/play#example/jsdoc-deprecated
```suggestion
 * @deprecated adaptor from async iterable to notifier.
```

Reply:
Also please explain here why it's deprecated.

> I have intentionally included the changes from https://github.com/Agoric/agoric-sdk/pull/5695 along with an immediate reversion for posterity.

I request that those two commits be omitted (or this be landed as squash). For master it's noise. If there's something important to convey about how we got to this point then just write it in HEAD. If you want there to be some record of the other work done, it's already in the referenced PR.

Reply:
I'm anticipating the attempt to come up again, and would like the prior record to outlive that dead branch.

Reply:
The branch will be dead but https://github.com/Agoric/agoric-sdk/pull/5695 will endure for anyone who wants to see it.

Non-blocking but I do think the PR suffices as a record and to the point of discover as to why this is the design, I don't think commit history suffices. If that's the goal is should be written into HEAD.